src/main/java/com/example/demo/AnotherConfig.java

```diff
@@ -25,4 +26,13 @@ public void setServletContext( ServletContext servletContext ) {
 		this.servletContext = servletContext;
 	}
 
-}
\ No newline at end of file
+	/**
+	 * I have a bean that depends on servletContext
+	 */
+	@Bean
```

Comment:
Why does this need to be a bean?

Reply:
I would need to use it in other parts of the application.

In CometD context this bean will be a `BayeuxServer`.

Reply:
Yeah that won't work for sure. Can you give this a try?

```java
public class DemoBean implements ApplicationListener<ApplicationStartedEvent> {

   ....

   @Override
    public void onApplicationEvent(ApplicationStartedEvent event) {
      ApplicationContext applicationContext = event.getApplicationContext();
      this.servletContext = ((WebApplicationContext) applicationContext).getServletcontext();
    }

}
```